sdk/servicebus/service-bus/src/receivers/sessionReceiver.ts

```diff
@@ -155,14 +144,38 @@ export class SessionReceiverImpl<ReceivedMessageT extends ReceivedMessage | Rece
     sessionOptions: CreateSessionReceiverOptions,
     retryOptions: RetryOptions = {}
   ): Promise<SessionReceiver<ReceivedMessageT>> {
+    context.isSessionEnabled = true;
+    if (sessionOptions.sessionId) {
+      sessionOptions.sessionId = String(sessionOptions.sessionId);
+
+      // Check if receiver for given session already exists
+      for (const [, ctx] of Object.entries(context.namespace.clientContexts)) {
+        if (
+          context.entityPath === ctx.entityPath &&
+          ctx.messageSessions[sessionOptions.sessionId] &&
+          ctx.messageSessions[sessionOptions.sessionId].isOpen()
+        ) {
+          const errorMessage = getOpenSessionReceiverErrorMsg(
+            ctx.entityPath,
+            sessionOptions.sessionId
+          );
+          const error = new Error(errorMessage);
+          log.error(`[${ctx.namespace.connectionId}] %O`, error);
+          throw error;
+        }
+      }
```

Comment:
In #9766, I am proposing we remove these checks altogether as they are no longer needed.
```suggestion
```